src/Microsoft.Health.Fhir.Shared.Client/FhirClient.cs

```diff
@@ -374,67 +389,26 @@ private async Task<FhirResponse<T>> CreateResponseAsync<T>(HttpResponseMessage r
                 string.IsNullOrWhiteSpace(content) ? null : (T)_deserialize(content));
         }
 
-        private async Task SetupAuthenticationAsync(TestApplication clientApplication, TestUser user = null)
+        private void SetupSecurity()
```

Comment:
We could also make this method public and have the caller explicitly initiate authentication process.